src/charm.py

```diff
@@ -129,6 +135,20 @@ def _on_update_status(self, _):
     # END: hook functions
 
     # BEGIN: helper functions
+    def get_keyfile_contents(self) -> str:
+        """Retrieves the contents of the keyfile on host machine."""
+        # wait for keyFile to be created by leader unit
+        if not self.get_secret(APP_SCOPE, Config.Secrets.SECRET_KEYFILE_NAME):
+            logger.debug("waiting to receive keyfile contents from config-server.")
```

Comment:
This check isn't needed for the charm to function. But the logs are helpful in case of error.

Let me first explain this function. This function "Retrieves the contents of the keyfile **on host machine.**" the reason we want to retrieve the contents from the host machine and **not** the secret is that we want to see what KeyFIle the mongos/mongod daemon is using. 

Just because its stored in the secret **does not** mean that the daemon is configured to read from the secret. The secret could be "behind/ahead" of what is written to the actual file. This is due to the nature of secret events. Its possible that something when wrong and the secret was removed and we could still be running with a KeyFile previously provided.

In these edge cases, it is possible that the charm breaks and we can trace back to this log. I hope this is helpful

Reply:
Thanks for clean explanation :pray: